src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1738,7 +1738,16 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
         linux_profile = ContainerServiceLinuxProfile(admin_username=admin_username, ssh=ssh_config)
 
     windows_profile = None
-    if windows_admin_username:
+    if windows_admin_username or windows_admin_password:
+        # To avoid that windows_admin_password is set but windows_admin_username is not
+        if windows_admin_username is None:
+            try:
+                windows_admin_username = prompt('windows_admin_username: ')
```

Comment:
To align with password below, it seems that it's better to use `windows-admin-username: ` here.